vignettes/createReposHtml.Rmd

```diff
@@ -0,0 +1,142 @@
+---
+title: "HOWTO generate repository HTML"
+author: "S. Falcon"
+date: "`r format(Sys.time(), '%B %d, %Y')`"
+vignette: >
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteIndexEntry{HOWTO generate repository HTML}
+  %\VignetteEncoding{UTF-8}
+output:
+  BiocStyle::html_document:
+    number_sections: yes
+    toc: yes
+    toc_depth: 4
+---
+
+
+# Overview
+
+This document assumes you have a collection of R packages on local disk that you would
+like to prepare for publishing to the web. The end result we are going for is:
+
+1. Packages organized per CRAN-style repository standard.
+
+2. PACKAGES files created for install.packages access.
+
+3. VIEWS file created for generating biocViews.
+
+4. A vignette directory created containing the extracted vignette pdf files from each source package in the repository.
+
+5. An html directory created containing html descriptions of each package with links for downloading available artifacts.
+
+6. A simple alphabetical listing index.html file.
+
+
+# CRAN-style Layout
+
+Establish a top-level directory for the repository, we will refer to this directory as reposRoot. Place your packages as follows:
+
+
+**src/contrib** Contains all source packages (*.tar.gz).
+
+**bin/windows/contrib/x.y** Contains all win.binary packages (*.zip). Where x.y is the major.minor version number of R.
+
+**bin/macosx/contrib/x.y** Contains the mac.binary (High Sierra) (*.tgz) packages.
+
+
+You will need the following parameters:
+
+```{r}
+## > reposRoot <- "path/to/reposRoot"
+## > ## The names are essential
+## > contribPaths <- c(source="src/contrib",
+## +                   win.binary="bin/windows/contrib/4.0",
+## +                   mac.binary="bin/macosx/contrib/4.0")
+```
+
+
+# Extracting vignettes
+
+The `extractVignettes` function extracts pdf files from inst/doc. The default is to extract to a reposRoot/vignettes.
+
+
+`> extractVignettes(reposRoot, contribPaths["source"])`
```

Comment:
This should be in a code block with 3 tick marks and `{r echo=TRUE, eval=FALSE}`.

Reply:
(Commenting here as a note to myself) - completed.